Tasks/Xcode/xcode.ts

```diff
@@ -363,6 +365,14 @@ async function run() {
                     }
 
                     if (xcodeVersion >= 9 && exportOptions === 'auto') {
+                        if (cloudEntitlement === true) {
+                            tl.debug("Adding cloud entitlement");
+                            if (exportMethod === "app-store") {
```

Comment:
This logic an move into the getCloudEntitlement helper and you can just check if(cloudEntitlement) and add the value to the exportOptionsPlist.